web/components/create-question-button.tsx

```diff
@@ -26,7 +28,12 @@ export const CreateQuestionButton = (props: {
         </Link>
       ) : (
         <button
-          onClick={firebaseLogin}
+          onClick={async () => {
+            // login, and then reload the page, to hit any SSR redirect (e.g.
+            // redirecting from / to /home for logged in users)
+            await firebaseLogin()
+            router.replace(router.asPath)
```

Comment:
This might look a little odd but it seems like it's the consensus for the best way to reload server-side props with Next right now. See https://github.com/vercel/next.js/discussions/12646.